cli/src/pcluster/models/cluster.py

```diff
@@ -12,939 +12,340 @@
+        self.name = name
+        self.stack = stack
+        self.bucket = None
+        self.template_body = None
+        self.config_version = None
+        if config:
+            self.config = ClusterSchema().load(config)
+        else:
+            try:
+                self.stack = ClusterStack(self.stack_name)
+            except StackActionError:
+                raise ClusterActionError(f"Cluster {self.name} does not exist.")
+            self.config = self.stack.get_cluster_config()
 
     @property
-    def mount_dir_list(self):
-        """Retrieve the list of mount dirs for the shared storage and head node ephemeral volume."""
-        mount_dir_list = []
-        if self.shared_storage:
-            for storage in self.shared_storage:
-                mount_dir_list.append(storage.mount_dir)
-
-        if self.head_node.storage and self.head_node.storage.ephemeral_volume:
-            mount_dir_list.append(self.head_node.storage.ephemeral_volume.mount_dir)
-
-        return mount_dir_list
-
-    @property
-    def compute_subnet_ids(self):
-        """Return the list of all compute subnet ids in the cluster."""
-        return list(
-            {
-                subnet_id
-                for queue in self.scheduling.queues
-                if queue.networking.subnet_ids
-                for subnet_id in queue.networking.subnet_ids
-                if queue.networking.subnet_ids
-            }
-        )
-
-    @property
-    def compute_security_groups(self):
-        """Return the list of all compute security groups in the cluster."""
-        return list(
-            {
-                security_group
-                for queue in self.scheduling.queues
-                if queue.networking.security_groups
-                for security_group in queue.networking.security_groups
-            }
-        )
+    def stack_name(self):
+        """Return stack name."""
+        return get_stack_name(self.name)
+
+    def create(self, disable_rollback: bool = False):
+        """Create cluster."""
+        # check cluster existence
+        if AWSApi.instance().cfn.stack_exists(self.stack_name):
+            raise ClusterActionError(f"Cluster {self.name} already exists")
+
+        validation_failures = self.config.validate()
+        if validation_failures:
+            # TODO skip validation errors
+            raise ClusterActionError("Configuration is invalid", validation_failures=validation_failures)
+
+        # Add tags information to the cluster
+        version = get_installed_version()
+        tags = self.config.tags or []
+        tags.append(Tag(key="Version", value=version))
+        tags = [{"Key": tag.key, "Value": tag.value} for tag in tags]
+
+        # Create bucket if needed
+        self._setup_cluster_bucket()
+
+        creation_result = None
+        try:
+            # Create template if not provided by the user
+            if not (self.config.dev_settings and self.config.dev_settings.cluster_template):
```

Comment:
The custom cluster_template is specified in the dev_settings section of the configuration file :D

BTW I understand the question. If cluster_template is provided we don't need to generate the template but the config file is required to know that you have a custom template and it might be useful because the custom template can have an user-data that uses the data from the configurafion file.

Reply:
The custom cluster_template is specified in the dev_settings section of the configuration file :D

BTW I understand the question. If cluster_template is provided we don't need to generate the template but the config file is required to know that you have a custom template and it might be useful because the custom template can have an user-data that uses the data from the configurafion file.